modules/ingester/instance.go

```diff
@@ -317,6 +314,23 @@ func (i *instance) resetHeadBlock() error {
 	return err
 }
 
+func (i *instance) tracesToCut(cutoff time.Duration, immediate bool) []*trace {
+	i.tracesMtx.Lock()
+	defer i.tracesMtx.Unlock()
+
+	now := time.Now()
+	tracesToCut := make([]*trace, 0, len(i.traces))
+
+	for key, trace := range i.traces {
+		if now.Add(cutoff).After(trace.lastAppend) || immediate {
```

Comment:
Maybe move the now.Add cutoff computation to the place where now is declared.

Reply:
Good call.  Thanks for the second set of eyes.